### PR TITLE
Alerting: Fix permissions for timeintervals UI and improve display of contact points in notification policies

### PR DESCRIPTION
**Pull request by tomratcliffe**
**What is this feature?**
* Improves granular access to time intervals (mute timings), which are displayed in a tab on the same page as notification policies. Fixes the permissions checks that are used to decide whether to show the `Notification policies` navtree item, and the associated checks in the UI code to redirect away if the user doesn't have access
* Fixes the error message on the policies page preventing a user from seeing Mute Timings. If they didn't have access to the necessary endpoints for the policies page, this would block being able to see the contents of the mute timings tab
* Removes some unnecessary props that were being passed through policies (as this is a recursively used component, it was easy to miss/spot that they weren't being used)
* Fixes some links from parts of the alerting UI -> contact points. We are always trying to link this using the contact point **name** directly to the view/edit page - but this doesn't work with the name when using `alertingApiServer`. In the cases where we don't have access to the contact point's UID, we link to the search page instead
* If the user cannot see the contact point that relates to a notification policy, then we don't try and link to it from that page (this will become more apparent when granular policy permissions are implemented)
* Add an empty state for the contact points page when you've filtered down to no results (in case we link from elsewhere -> the list view but the user can't see the contact point in question)
* Improves the preview/popup for the contact point for a given policy. We now reuse the existing `ContactPointReceiverSummary` component that is used in the dropdowns

Before: 
image

After (this will be enriched further when changes are made to policies page to support k8s API):
image

Showing "no integrations" in the dropdown list:
image

### CODE REVIEW

File: pkg/services/navtree/navtreeimpl/navtree.go [hunk 1]
```diff
@@ -448,6 +448,8 @@ func (s *ServiceImpl) buildAlertNavLinks(c *contextmodel.ReqContext) *navtree.Na
 	if hasAccess(ac.EvalAny(
 		ac.EvalPermission(ac.ActionAlertingNotificationsRead),
 		ac.EvalPermission(ac.ActionAlertingNotificationsExternalRead),
+		ac.EvalPermission(ac.ActionAlertingNotificationsTimeIntervalsRead),
```
Comment (tomratcliffe) [inline pkg/services/navtree/navtreeimpl/navtree.go:451]: (So we show the Notification Policies navtree item if a user only has permissions to see Mute Timings, but not Policies)
image

File: public/app/features/alerting/unified/components/mute-timings/permissions.ts
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/mute-timings/permissions.ts:1]: 👍 love this pattern

File: public/app/features/alerting/unified/NotificationPolicies.tsx [hunk 2]
```diff
@@ -242,54 +242,51 @@ const AmRoutes = () => {
       </TabsBar>
       <TabContent className={styles.tabContent}>
         {isFetching && <LoadingPlaceholder text="Loading Alertmanager config..." />}
-        {haveError && (
-          <Alert severity="error" title="Error loading Alertmanager config">
-            {resultError.message || 'Unknown error.'}
-          </Alert>
-        )}
-        {haveData && (
```
[the remark this answers is not in the document]
Comment (gillesdemey) [reply public/app/features/alerting/unified/NotificationPolicies.tsx:250]: Good idea to decouple those; historically those would have loaded from the same config endpoint but with our move to the new APIs this should definitely get tackled.

File: public/app/features/alerting/unified/NotificationPolicies.tsx [hunk 3]
```diff
@@ -242,54 +242,51 @@ const AmRoutes = () => {
...
+                    provisioned={isProvisioned}
+                    alertManagerSourceName={selectedAlertmanager}
+                    onAddPolicy={openAddModal}
+                    onEditPolicy={openEditModal}
+                    onDeletePolicy={openDeleteModal}
+                    onShowAlertInstances={showAlertGroupsModal}
+                    routesMatchingFilters={routesMatchingFilters}
+                    matchingInstancesPreview={{ groupsMap: routeAlertGroupsMap, enabled: !instancesPreviewError }}
+                    isAutoGenerated={false}
+                    isDefaultPolicy
```
[the remark this answers is not in the document]
Comment (gillesdemey) [reply public/app/features/alerting/unified/NotificationPolicies.tsx:276]: Yea, only the very first (top-level) policy is the "default" one

File: public/app/features/alerting/unified/components/contact-points/ContactPoint.tsx [hunk 4]
```diff
@@ -176,6 +170,11 @@ export const ContactPointReceiverSummary = ({ receivers, limit }: ContactPointRe
   return (
     <Stack direction="column" gap={0}>
       <Stack direction="row" alignItems="center" gap={1}>
+        {integrationsShown.length === 0 && (
+          <MetaText color="warning" icon="exclamation-circle">
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/contact-points/ContactPoint.tsx:174]: nit: for warnings we tend to use `exclamation-triangle`, though I do realize this was just code that was moved ;)

File: public/app/features/alerting/unified/components/notification-policies/Policy.tsx [hunk 5]
```diff
@@ -894,21 +872,13 @@ const ContactPointsHoverDetails: FC<ContactPointDetailsProps> = ({
...
-          {/* use "label" to indicate how many of that type we have in the contact point */}
-          {Object.entries(groupedIntegrations).map(([type, integrations]) => (
-            <Label
-              key={uniqueId()}
-              label={integrations.length > 1 ? integrations.length : undefined}
-              icon={INTEGRATION_ICONS[type]}
-              value={upperFirst(type)}
-            />
-          ))}
+        <Stack justifyContent="center">
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/notification-policies/Policy.tsx]: It looks a bit odd being centered, I'll play around with the card design a bit to see what could work. What if we made the header smaller too?

image

File: public/app/features/alerting/unified/hooks/useAbilities.ts [hunk 6]
```diff
@@ -331,7 +351,8 @@ function useCanSilence(rule: CombinedRule): [boolean, boolean] {
...
 
 // just a convenient function
-const toAbility = (supported: boolean, ...actions: Array<AccessControlAction | null>): Ability => [
-  supported,
-  actions.some((action) => action && ctx.hasPermission(action)),
-];
+const toAbility = (
+  supported: boolean,
+  /** If user has any of these permissions, then they are allowed to perform the action */
+  ...actions: Array<AccessControlAction | null>
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/hooks/useAbilities.ts:357]: With the changes above I think we should be able to drop `| null`

**Review by gillesdemey — APPROVED** [review]
Changes LGTM, just a few nits

**Pull request merged**